linux/hid.c

```diff
@@ -639,15 +690,19 @@ hid_device * HID_API_EXPORT hid_open_path(const char *path)
 
 		/* Get Report Descriptor Size */
 		res = ioctl(dev->device_handle, HIDIOCGRDESCSIZE, &desc_size);
-		if (res < 0)
-			perror("HIDIOCGRDESCSIZE");
-
+		if (res < 0) {
+			char errormsg[100];
+			snprintf(errormsg, sizeof(errormsg), "ioctl (GRDESCSIZE): %s", strerror(errno));
+			register_device_error(dev, errormsg);
```

Comment:
I'm not very fond of this construct being present in multiple places. Should I create a helper function for that? If yes, name suggestions?

Reply:
`register_device_ioctl_error`?

Reply:
I've gone for `static void register_device_error_format(hid_device *dev, const char *format, ...)`